### PR TITLE
Fix infinite frequency problem

### PR DESCRIPTION
**Pull request by luojing1211**
This is a fix for the infinite frequency problem for FD model.

### CODE REVIEW

**Comment by luojing1211** [comment]
@paulray Can you try it out now?

**Comment by paulray** [comment]
I think the same fix needs to be applied in `d_delay_FD_d_FDX()`

**Comment by paulray** [comment]
But, for my simple use case, this does seem to fix it. Thanks!

**Comment by paulray** [comment]
Ideally, it would be good to have a unit test with an infinite freq TOA

**Comment by luojing1211** [comment]
Hi Paul, 
I just pushed the fix for derivatives and added some infinite frequency test.

File: tests/test_fd.py [hunk 1]
```diff
@@ -28,6 +29,19 @@ def test_FD(self):
         # Those two clock correction difference are causing the trouble.
         assert np.all(resDiff< 5e-6) , \
             "PINT and tempo Residual difference is too big. "
+    def test_inf_freq(self):
+        test_toas = copy.deepcopy(self.toas)
+        test_toas.table['freq'][0:5] = np.nan
```
Comment (paulray) [inline tests/test_fd.py]: I think this should be np.inf * u.MHz (i.e. it should not be np.nan and it should have units)

Comment (luojing1211) [inline tests/test_fd.py]: It is fixed now.

**Pull request merged**